kymatio/scattering2d/backend/backend_skcuda.py

```diff
@@ -69,23 +69,25 @@ def __init__(self, pad_size, input_size, pre_pad=False):
         self.build()
 
     def build(self):
-        pad_size_tmp = self.pad_size
+        pad_size_tmp = list(self.pad_size)
 
         # This allow to handle the case where the padding is equal to the image size
-        if pad_size_tmp[0] == self.input_size[0]:
-            pad_size_tmp[0] -= 1
-            pad_size_tmp[1] -= 1
-        if pad_size_tmp[2] == self.input_size[1]:
+        # We need to take care that the order of the pad sizes is reversed with
+        # respect to the input size.
+        if pad_size_tmp[2] == self.input_size[0]:
             pad_size_tmp[2] -= 1
             pad_size_tmp[3] -= 1
+        if pad_size_tmp[0] == self.input_size[1]:
+            pad_size_tmp[0] -= 1
```

Comment:
For my own edification: I have been wondering through a bunch of potential reasons for the `-1` here. What is the actual reason?

Reply:
(My No. 1: Because it's reflection padding and you don't want to copy the one right at the border?)

Reply:
Actually, I'd prefer this to error instead of making such a modification (which will confuse users, who will be left with the feeling that the software doesn't do what they asked of it.). Either error or continue with a new reflection beyond the first reflection.

Reply:
The problem is that reflection padding isn't defined for a padding equal to the input size: you can extend `0 1 2` to `2 1 | 0 1 2 | 1 0` but no further. So in order to add that extra padding we manually do a second reflection: `1 2 1 | 0 1 2 | 2 1 2`.

Reply:
Sounds good to me